crates/context_aware_config/src/api/dimension/handlers.rs

```diff
@@ -86,39 +93,155 @@ async fn create(
+                dimensions::position.eq(dimensions::position + 1),
+            ))
+            .execute(transaction_conn)?;
+        let insert_resp = diesel::insert_into(dimensions)
+            .values(&dimension_data)
+            .get_result::<Dimension>(transaction_conn);
 
-    match upsert {
-        Ok(upserted_dimension) => {
-            let is_mandatory = tenant_config
-                .mandatory_dimensions
-                .contains(&upserted_dimension.dimension);
-            Ok(HttpResponse::Created().json(DimensionWithMandatory::new(
-                upserted_dimension,
-                is_mandatory,
-            )))
+        match insert_resp {
+            Ok(inserted_dimension) => {
+                let is_mandatory = tenant_config
+                    .mandatory_dimensions
+                    .contains(&inserted_dimension.dimension);
+                Ok(HttpResponse::Created().json(DimensionWithMandatory::new(
+                    inserted_dimension,
+                    is_mandatory,
+                )))
+            }
+            Err(diesel::result::Error::DatabaseError(
+                diesel::result::DatabaseErrorKind::ForeignKeyViolation,
+                e,
+            )) => {
+                log::error!("{fun_name:?} function not found with error: {e:?}");
+                Err(bad_argument!(
+                    "Function {} doesn't exists",
+                    fun_name.unwrap_or(String::new())
+                ))
+            }
+            Err(e) => {
+                log::error!("Dimension create failed with error: {e}");
+                Err(unexpected_error!(
+                    "Something went wrong, failed to create dimension"
+                ))
+            }
         }
-        Err(diesel::result::Error::DatabaseError(
-            diesel::result::DatabaseErrorKind::ForeignKeyViolation,
-            e,
-        )) => {
-            log::error!("{fun_name:?} function not found with error: {e:?}");
-            Err(bad_argument!(
-                "Function {} doesn't exists",
-                fun_name.unwrap_or(String::new())
-            ))
+    })
+}
+
+#[put("/{name}")]
+async fn update(
+    path: Path<DimensionName>,
+    state: Data<AppState>,
+    req: web::Json<UpdateReq>,
+    user: User,
+    db_conn: DbConnection,
+    tenant_config: TenantConfig,
+) -> superposition::Result<HttpResponse> {
+    let name: String = path.clone().into();
+    use dimensions::dsl;
+    let DbConnection(mut conn) = db_conn;
+
+    let mut dimension_row: Dimension = dsl::dimensions
+        .filter(dimensions::dimension.eq(name.clone()))
+        .get_result::<Dimension>(&mut conn)?;
+
+    let update_req = req.into_inner();
+
+    if let Some(schema_value) = update_req.schema {
+        validate_jsonschema(&state.meta_schema, &schema_value)?;
+
+        let schema_compile_result = JSONSchema::options()
+            .with_draft(Draft::Draft7)
+            .compile(&schema_value);
```

Comment:
this JSONSchema creation is common in a lot of places along with `validate_jsonschema`, should we not couple these to avoid missing out either of this